File: 14-04-2024/Xoring and Clearing.js
//{ Driver Code Starts
// Initial Template for javascript
// Position this line where user code will be pasted.

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split("\n").map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let n = parseInt(readLine());

        let arr = [];
        let input_arr = readLine().split(' ').map(Number);
        for (let j = 0; j < n; j++) {
            arr[j] = input_arr[j];
        }

        let obj = new Solution();
        obj.xor1ToN(n, arr);
        obj.printArr(n, arr);
        obj.setToZero(n, arr);
        obj.printArr(n, arr);
        // console.log(res);
    }
}

// } Driver Code Ends




// User function Template for javascript

class Solution {
    printArr(n, arr) {
        //Your Code Start
        // Your code for printing array here
           let result = "";
        for (let i = 0; i < n; i++) {
            result += arr[i];
            if (i < n - 1) {
                result += " ";
            }
        }
        console.log(result);
    }

    setToZero(n, arr) {
         for(let i = 0; i < n; i++) {
            arr[i] = 0;
        }
        // set element of arr to zero
    }

    xor1ToN(n, arr) {
        for(let i = 0; i < n; i++) {
            arr[i] ^= i;
        }
        // Xor arr[i]^i
        
        //Your Code End
    }
}

File: 14-04-2024/Xoring and Clearing.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
void printArr(int n, int arr[]) {
    //Your Code Start
        // Your code for printing array here
        for(int i=0;i<n;i++){
            cout<<arr[i]<<" ";
        }
        cout<<endl;
    }

    void setToZero(int n, int arr[]) {
        // Use memset to set arr to zero
        memset(arr,0,sizeof(int)*n);
    }

    void xor1ToN(int n, int arr[]) {
        // Xor arr[i]^i
        for(int i=0;i<n;i++){
            arr[i]^=i;
        }
        //Your Code End
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution obj;
        obj.xor1ToN(n, arr);
        obj.printArr(n, arr);
        obj.setToZero(n, arr);
        obj.printArr(n, arr);
    }
    return 0;
}

// } Driver Code Ends
